lib/manager/gomod/artifacts.ts

```diff
@@ -152,10 +152,21 @@ export async function updateArtifacts({
   const useVendor = (await readLocalFile(vendorModulesFileName)) !== null;
 
   try {
-    const massagedGoMod = newGoModContent.replace(
-      regEx(/\n(replace\s+[^\s]+\s+=>\s+\.\.\/.*)/g),
-      '\n// renovate-replace $1'
+    const inlineReplaceRegEx: RegExp = regEx(
+      /\n(replace\s+[^\s]+\s+=>\s+\.\.\/.*)/g
     );
+
+    const inlineCommentOut = '\n// renovate-replace $1';
+
+    const blockReplaceRegEx: RegExp = regEx(/\nreplace\s*\([^)]+\s*\)/g);
+
+    const blockCommentOut = (match): string =>
+      match.replaceAll('\n', '\n// renovate-replace ');
```

Comment:
```suggestion
      match.replaceAll('(\r?\n)', '$1// renovate-replace ');
```

Reply:
added all the necessary changes as suggested.

does it really matter if we include the \r in the regex, as it is not effected by the regex prior to the suggested changes, as it is untouched if any exists. would love to know as i might have some misconceptions regarding regex :).